cluster-health.go

```diff
@@ -193,7 +194,7 @@ func (an *AnonymousClient) alive(ctx context.Context, u *url.URL, resource strin
 			dnsDoneTime = time.Now()
 		},
 		GetConn: func(_ string) {
-			reqStartTime = time.Now()
+			reqStartTime = timeNow
```

Comment:
Maybe what you can do is
```
if reqStartTime.IsZero() {
   reqStartTime = time.Now()
}
```

and add this link as comment, https://github.com/golang/go/issues/44281